impl/game_server_lib/validity_checker.cpp

```diff
@@ -0,0 +1,31 @@
+#include "validity_checker.hpp"
+#include "map/terrain.hpp"
+
+bool ValidityChecker::checkIfUnitIDsAreUnique(
+    std::vector<UnitServerRoundStartData> const& existingUnits,
+    UnitClientToServerData const& newUnitData)
+{
+    for (auto const& unitData : existingUnits) {
+        if (unitData.unitClientToServerData.unitID == newUnitData.unitID
+            && unitData.unitClientToServerData.playerID == newUnitData.playerID) {
+
+            return false;
+        }
+    }
+    return true;
+}
+bool ValidityChecker::checkIfUnitPlacementIsValid(UnitClientToServerData const& newUnitData)
+{
+    if (newUnitData.positionX < 0) {
+        return false;
+    } else if (newUnitData.positionX >= terrainWidthInChunks * terrainChunkSizeInPixel) {
+        return false;
+    }
```

Comment:
```suggestion
    if (newUnitData.positionX < 0) {
        return false;
    }
    if (newUnitData.positionX >= terrainWidthInChunks * terrainChunkSizeInPixel) {
        return false;
    }
```
No else needed, after return? 😄